packages/client/src/schedule-types.ts

```diff
@@ -0,0 +1,845 @@
+     * Window could lead to missed Actions. (But you can always {@link ScheduleHandle.backfill}.)
+     *
+     * @default 1 minute
+     * @format number of milliseconds or {@link https://www.npmjs.com/package/ms | ms-formatted string}
+     */
+    catchupWindow?: number | string;
+
+    /**
+     * When an Action times out or reaches the end of its Retry Policy, {@link pause}.
+     *
+     * With {@link ScheduleOverlapPolicy.ALLOW_ALL}, this pause might not apply to the next Action, because the next Action
+     * might have already started previous to the failed one finishing. Pausing applies only to Actions that are scheduled
+     * to start after the failed one finishes.
+     *
+     * @default false
+     */
+    pauseOnFailure?: boolean;
+  };
+
+  /**
+   * Additional non-indexed information attached to the Schedule. The values can be anything that is
+   * serializable by the {@link DataConverter}.
+   */
+  memo?: Record<string, unknown>;
+
+  /**
+   * Additional indexed information attached to the Schedule. More info:
+   * https://docs.temporal.io/docs/typescript/search-attributes
+   *
+   * Values are always converted using {@link JsonPayloadConverter}, even when a custom Data Converter is provided.
+   */
+  searchAttributes?: SearchAttributes;
+
+  /**
+   * The initial state of the schedule, right after creation or update.
+   */
+  state?: {
+    /**
+     * Start in paused state.
+     *
+     * @default false
+     */
+    paused?: boolean;
+
+    /**
+     * Informative human-readable message with contextual notes, e.g. the reason
+     * a Schedule is paused. The system may overwrite this message on certain
+     * conditions, e.g. when pause-on-failure happens.
+     */
+    note?: string;
+
+    /**
+     * Limit the number of Actions to take.
+     *
+     * This number is decremented after each Action is taken, and Actions are not
+     * taken when the number is `0` (unless {@link ScheduleHandle.trigger} is called).
+     *
+     * If `undefined`, then no such limit applies.
+     *
+     * @default undefined, which allows for unlimited exections
+     */
+    remainingActions?: number;
+
+    /**
+     * Trigger one Action immediately on create.
+     *
+     * @default false
+     */
+    triggerImmediately?: boolean;
+
+    /**
+     * Runs though the specified time periods and takes Actions as if that time passed by right now, all at once. The
+     * overlap policy can be overridden for the scope of the backfill.
+     */
+    backfill?: Backfill[];
+  };
+}
+
+/** @experimental */
+export type CompiledScheduleOptions = Replace<
+  ScheduleOptions,
+  {
+    action: CompiledScheduleAction;
+  }
+>;
+
+/**
+ * The specification of an updated Schedule, as expected by {@link ScheduleHandle.update}.
+ *
+ * @experimental
+ */
+export type ScheduleUpdateOptions = Omit<ScheduleOptions, 'scheduleId' | 'memo' | 'searchAttributes'>;
```

Comment:
Actually, removed `triggerImmediately` and `backfill` from ScheduleUpdateOptions.state as they are valid on create, but have to be updated through a patch rather than through `Update`